### ISSUE TITLE
Is there room for improvement in the GenServer Tutorial?

### ISSUE THREAD

**Issue by jaysoifer** [elixir-lang/elixir-lang.github.com#975]
On the section [Our first GenServer](http://elixir-lang.org/getting-started/mix-otp/genserver.html#our-first-genserver), there is an explanation that I feel like it could be improved:

• • • INTRO • • •

The first function is start_link/3, which starts a new GenServer passing three arguments:

 1. The module where the server callbacks are implemented, in this case __MODULE__, meaning the current module

 2. The initialization arguments, in this case the atom :ok

 3. A list of options which can be used to specify things like the name of the server. For now, we pass an empty list

• • • ISSUE • • •

_(...) That’s it for the client API. On the server side, (...)_

**The first is the init/1 callback, that receives the argument given to GenServer.start_link/3 (...)**

• • •

So, I'm trying to understand how does `init/1` receives the argument given to another function that has an arity of 3 (`GenServer.start_link/3`).

I'm not sure I got this right, is the argument received the item #2 on the above list? ("in this case the atom :ok").

If that is so, could it possibly be worthwhile to make this clearer?

**Comment by whatyouhide** [comment]
> is the argument received the item #2 on the above list?

Yep. `GenServer.start_link/3` spawns a new GenServer, then the `module.init/1` callback is called where `module` is the first argument to `start_link/3`. `module.init/1` is called with the second argument passed to `start_link/3`. If you want to cram more stuff into that single argument, it's common to use tuples or lists:

```elixir
def start_link() do
  GenServer.start_link(__MODULE__, {arg1, arg2, arg3})
end

def init({arg1, arg2, arg3}) do
  ...
end
```

I'm not sure this can be made clearer in the guide. Maybe the atom `:ok` is confusing, I don't know.

**Comment by jaysoifer** [comment]
I would suggest changing the part where it said:

_The first is the init/1 callback, that receives the argument given to GenServer.start_link/3 (...)_

to 

_The first is the init/1 callback, that receives the *second* argument given to GenServer.start_link/3 (...)_

Does that make sense?

**Comment by whatyouhide** [comment]
@jaysoifer yes it does :). Can you please send a PR? 💟

**Comment by jaysoifer** [comment]
Done 😃

**Comment by jaysoifer** [comment]
![hacker](https://cloud.githubusercontent.com/assets/8316698/25065735/37af4bfa-21eb-11e7-829a-3debf527108e.jpg)

**Comment by whatyouhide** [comment]
Closed by the much assembly in #976.
